Formula/sourcedocs.rb

```diff
@@ -0,0 +1,21 @@
+class Sourcedocs < Formula
+  last_release = "0.5.1"
+
+  desc "Generate Markdown files from inline source code documentation"
+  homepage "https://github.com/eneko/SourceDocs"
+  url "https://github.com/eneko/SourceDocs/archive/#{last_release}.tar.gz"
+  sha256 "3c2e2de695d49dbdd5acb49f8876042bdc97e8d6b95584d3ef6b592b8f10affc"
+
+  depends_on :xcode => ["9.3", :build]
+
+  def install
+    ohai "Building SourceDocs"
+    system "swift", "build", "--disable-sandbox", "-c", "release", "-Xswiftc", "-static-stdlib"
+    ohai "Installing SourceDocs..."
+    bin.install ".build/release/sourcedocs"
+  end
+
+  test do
+    assert_equal `#{bin}/sourcedocs version`.strip, "SourceDocs v#{last_release}"
```

Comment:
By exercising a deeper level of the functionality, we mean to test something the formula really does, i.e. generating some Markdown files in this particular case per the formula description.

Reply:
Hi @lembacon, thanks for the feedback. I was working on this last night, but having some trouble.

In order to test the Markdown generator, `sourcedocs` needs to be able to build a Swift module to process the type definitions for the documentation. Here is how my tests look right now:
```ruby
  test do
    assert_equal `#{bin}/sourcedocs version`.strip, "SourceDocs v0.5.1"

    system "mkdir", "foo"
    Dir.chdir('foo') {
      system "swift", "package", "init"
      system "swift", "build"
      system "sourcedocs", "generate", "--spm-module", "foo"
    }
    assert_true File.file?("foo/Documentation/Reference/README.md")
  end
```

The first two steps, creating a directory and initializing a Swift package, seem to work well. However, `swift build` fails with this error:
```
Last 15 lines from /Users/enekoalonso/Library/Logs/Homebrew/sourcedocs/test.03.swift:
2018-09-29 19:36:06 -0700

swift
build

/private/tmp/sourcedocs-test-20180929-91502-lyiw2u/foo: error: manifest parse error(s):
sandbox-exec: sandbox_apply_container: Operation not permitted
Error: sourcedocs: failed
undefined method `map' for nil:NilClass
Did you mean?  tap
/usr/local/Homebrew/Library/Homebrew/exceptions.rb:363:in `initialize'
/usr/local/Homebrew/Library/Homebrew/utils/fork.rb:13:in `new'
/usr/local/Homebrew/Library/Homebrew/utils/fork.rb:13:in `rewrite_child_error'
/usr/local/Homebrew/Library/Homebrew/utils/fork.rb:84:in `block (3 levels) in safe_fork'
/usr/local/Homebrew/Library/Homebrew/utils.rb:402:in `ignore_interrupts'
/usr/local/Homebrew/Library/Homebrew/utils/fork.rb:60:in `block (2 levels) in safe_fork'
/usr/local/Homebrew/Library/Homebrew/utils/fork.rb:29:in `open'
/usr/local/Homebrew/Library/Homebrew/utils/fork.rb:29:in `block in safe_fork'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/tmpdir.rb:89:in `mktmpdir'
/usr/local/Homebrew/Library/Homebrew/utils/fork.rb:28:in `safe_fork'
/usr/local/Homebrew/Library/Homebrew/dev-cmd/test.rb:81:in `block in test'
/usr/local/Homebrew/Library/Homebrew/dev-cmd/test.rb:29:in `each'
/usr/local/Homebrew/Library/Homebrew/dev-cmd/test.rb:29:in `test'
/usr/local/Homebrew/Library/Homebrew/brew.rb:91:in `<main>'
```

Any help would be appreciated.

Reply:
The line `sandbox-exec: sandbox_apply_container: Operation not permitted` seems to indicate there is a protected sandbox when testing. What would be the appropriate way to build the Swift module within the sandbox?